smart-send-logistics/smart-send-extend-woo-core.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+
+include __DIR__ . '/includes/class-ss-shipping-wc-order.php';
+
+/**
+ * Class Smart_Send_Extend_Woo_Core
+ *
+ * Extends the WooCommerce core functionality for the Smart Send Logistics plugin.
+ *
+ * This class hooks into the WooCommerce Store API to extend the checkout schema and save custom
+ * shipping instructions to the order metadata. It also manages session initialization for storing
+ * shipping agent data.
+ *
+ * @package    SmartSendLogistics
+ * @subpackage WooCommerce Integration
+ * @category   WooCommerce API Data Storing
+ * @author     Smart Send
+ */
+
+class Smart_Send_Extend_Woo_Core
+{
+
+	/**
+	 * Action hook for saving shipping instructions.
+	 *
+	 * @var string
+	 */
+	const ACTION_SAVE_SHIPPING_INSTRUCTIONS = 'woocommerce_store_api_checkout_update_order_from_request';
+
+	/**
+	 * Bootstraps the class and hooks required data.
+	 * 
+	 * @return void
+	 */
+	public function init()
+	{
+		$this->save_shipping_instructions();
+	}
+
+	/**
+	 * Register  schema into the Checkout endpoint.
+	 *
+	 * @return array Registered schema.
+	 */
+	public function extend_checkout_schema()
+	{
+		return [
+			'selectedPickupPoint' => [
+				'description' => 'Selected shipping pick-up point',
+				'type'        => 'string',
+				'context'     => ['view', 'edit'],
+				'readonly'    => true,
+				'arg_options' => [
+					'validate_callback' => function ($value) {
+						return is_string($value);
+					},
+				],
+			],
+		];
+	}
+
+	/** * Initializes session if not already started. */
+	private function initialize_session()
+	{
+		if (session_status() === PHP_SESSION_NONE) {
+			session_start();
+		}
+		if (!isset($_SESSION['initialized'])) {
+			session_regenerate_id(true);
+			$_SESSION['initialized'] = true;
+		}
+	}
+
+	/**
+	 * Saves the shipping instructions to the order's metadata.
```

Comment:
It is not shipping instructions, it is pickup points.

Reply:
updated.